### PR TITLE
Add search size window parameter to MRR and DCG metric

### PR DESCRIPTION
**Pull request by cbuescher**
In a similar vein as #27569, this PR adds a search window size parameter to the two other ranking evaluation metrics.

### CODE REVIEW

**Review by jimczi — APPROVED** [review]
LGTM

**Comment by cbuescher** [comment]
@mayya-sharipova I hope you don't mind I merged this to the main feature branch aleady, there were some test fixes in here I wanted to have running in CI. Feel free to leave any comments you have either here, or better in #27478.

**Comment by mayya-sharipova** [comment]
@cbuescher Christoph, it is good that you merged without waiting for any approval from me. I don't think I have enough knowledge yet to approve your PR, most learning and trying to understand :)

**Pull request merged**
